distbench_engine.cc

```diff
@@ -1247,8 +1253,16 @@ void DistBenchEngine::StartOpenLoopIteration(ActionState* action_state) {
       action_state->action->proto.iterations().open_loop_interval_ns());
   auto it_state = std::make_shared<ActionIterationState>();
   it_state->action_state = action_state;
+  auto& interval_distribution = action_state->action->proto.iterations()
+                                      .open_loop_interval_distribution();
   action_state->iteration_mutex.Lock();
-  action_state->next_iteration_time += period;
+  if(interval_distribution == "exponential_distribution"){
```

Comment:
avoid testing a string for equality in the inner loop, test it once, and store the result in the class.